nav2_map_server/src/map_3d/map_io_3d.cpp

```diff
@@ -0,0 +1,367 @@
+
+    load_parameters = loadMapYaml(yaml_file);
+  } catch (YAML::Exception & e) {
+    std::cerr << "[ERROR] [map_io_3d]: Failed processing YAML file " << yaml_file <<
+      " at position (" << e.mark.line << ":" << e.mark.column << ") for reason: " <<
+      e.what() << std::endl;
+
+    return LOAD_MAP_STATUS::INVALID_MAP_METADATA;
+  }
+
+  try {
+    loadMapFromFile(load_parameters, map_msg);
+  } catch (std::exception & e) {
+    std::cerr <<
+      "[ERROR] [map_io_3d]: Failed to load image file " << load_parameters.pcd_file_name <<
+      " for reason: " << e.what() << std::endl;
+
+    return LOAD_MAP_STATUS::INVALID_MAP_DATA;
+  }
+
+  return LOAD_MAP_STATUS::LOAD_MAP_SUCCESS;
+}
+
+// === Map output part ===
+
+/**
+ * @brief Checks map saving parameters for consistency
+ * @param save_parameters Map saving parameters.
+ * NOTE: save_parameters could be updated during function execution.
+ * @throw std::exception in case of inconsistent parameters
+ */
+void checkSaveParameters(SaveParameters & save_parameters)
+{
+  // Check for empty file name/path
+  if (save_parameters.map_file_name.empty()) {
+    rclcpp::Clock clock(RCL_SYSTEM_TIME);
+    save_parameters.map_file_name =
+      "map_" + std::to_string(static_cast<int>(clock.now().seconds()));
+    std::cout << "[WARNING] [map_io_3d]: Map file unspecified. Map will be saved to " <<
+      save_parameters.map_file_name << " file" << std::endl;
+  }
+
+  // Check for encoding
+  if (save_parameters.as_binary) {
+    std::cout << "[WARNING] [map_io_3d]: Map will be saved in binary form to " <<
+      save_parameters.map_file_name << " file" << std::endl;
+  }
+
+  // Check for file format
+  if (save_parameters.format.empty()) {
+    save_parameters.format = "pcd";
+    std::cout << "[WARNING] [map_io_3d]: No map format is "
+      "specified. pcd format will be used." << std::endl;
+  }
+
+  // Confirm for the presently implemented formats
+  if (save_parameters.format != "ply" && save_parameters.format != "pcd") {
+    save_parameters.format = "pcd";
+    std::cout << "[WARNING] [map_io_3d]: " << save_parameters.format <<
+      " support is not implemented, Falling back to pcd file format" << std::endl;
+  }
+
+  if (save_parameters.format == "ply") {
+    // TODO(Shivam Pandey): add ply support
+    save_parameters.format = "pcd";
+    std::cout << "[WARNING] [map_io_3d]: ply support is not implemented, "
+      "Falling back to pcd file format" << std::endl;
+  }
+}
+
+/**
+ * @brief Tries to write map data into a file
+ * @param map Pointcloud2 data
+ * @param save_parameters Map saving parameters
+ * @throw std::exception in case of problem
+ */
+void tryWriteMapToFile(
+  const sensor_msgs::msg::PointCloud2 & map,
+  const SaveParameters & save_parameters)
+{
+  std::string file_name(save_parameters.map_file_name);
+
+  if (save_parameters.format == "pcd") {
+    file_name += ".pcd";
+  } else if (save_parameters.format == "ply") {
+    file_name += ".ply";
+  }
```

Comment:
This is excess code: PLY is not implemented yet and in `checkSaveParameters()` this intentionally being fixed to `pcd`. Please remove it.